### ISSUE TITLE
Upgrade embedded Kotlin to 1.7.21

### ISSUE THREAD

**Issue by eskatos** [gradle/gradle#22859]
See

* https://github.com/JetBrains/kotlin/releases/tag/v1.7.21

Blocked on

* https://github.com/gradle/gradle/issues/22853

**Comment by eskatos** [comment]
Superseded by 

* https://github.com/gradle/gradle/issues/22101
